### DISCUSSION TITLE
Can't use my custom api correctly

### DISCUSSION THREAD

**Discussion by comrade1059** [brianpetro/obsidian-smart-connections#914] [Q&A]
Firstly, after two days of experience (thus exhausting the official OpenAI API quota),I feel that the new version has significantly improved stability and user comfort. Thank you to the developers for contributions. 
I filled out the content identically in both the custom API of version 2.2.85 and version 2.3.40. However, while I could normally call the API in the former, the latter displayed "no model selected" and was unusable. The specific details are as follows (Figure 1 shows the situation in version 2.2.85, Figure 2 shows the situation in version 2.3.40, and Figure 3 displays the console content).
Once again, thank you!
![图1](https://github.com/user-attachments/assets/d5aa3431-dee6-40a9-8cd3-090a22ac44da)
![图2](https://github.com/user-attachments/assets/1d3910fd-c8c9-4164-88c5-b586e2b5f328)
![图3](https://github.com/user-attachments/assets/87f389aa-7493-4a68-8362-7a893775ca9c)

I tried to add "model_key" and "id" in the smart_en.json's custom, and the error "Selected model does not support tools"(but i think this error seems not to matter) instead of "no model selected".  The more important is console show this: 
![图4](https://github.com/user-attachments/assets/08921223-404b-40c6-ade5-e88ab045c291)
I have no idea about it, though i've tried some methods as much as i can.

Thank you again!

**Comment by lynic** [comment]
Same here, getting same error msg "No model selected."

**Comment by brianpetro** [comment]
See https://github.com/brianpetro/obsidian-smart-connections/issues/921#issuecomment-2540394360 🌴

**Comment by yamogu** [comment]
Was this fixed for everybody? Still can't use custom model.

**Reply by brianpetro** [reply]
Which platform are you trying to use?
